__tests__/end-to-end.js

```diff
@@ -93,18 +125,33 @@ function makeMakeTest (defaultDependentTests: Array<string> | string = []) {
       }
 
       // first make sure all dependent tests have passed
-      if (!(dependentTests instanceof Array)) {
-        dependentTests = [dependentTests]
-      }
-      dependentTests = [...defaultDependentTests, ...dependentTests]
-
+      // $FlowFixMe, should be an array by now
       dependentTests.forEach(test => {
         if (!testResults[test]) {
           log.error(`Dependent test "${test}" has not completed yet`)
           throw new Error(`Dependent test "${test}" has not completed yet`)
         }
       })
 
+      // if the ISOLATED_TEST is set, only run a test if it is the isolated test
+      // or if it is a test that the isolated test depends on passing
+      if (ISOLATED_TEST) {
+        if (!testDependencies[ISOLATED_TEST]) {
+          throw new Error(`Isolated test not defined: "${ISOLATED_TEST}"`)
+        }
+
+        if (
+          !(
+            name === ISOLATED_TEST ||
+              testDependencies[ISOLATED_TEST].indexOf(name) > -1
+          )
```

Comment:
Would prefer this conditional to read:
```
if (name !== ISOLATED_TEST && testDependencies[ISOLATED_TEST].indexOf(name) === -1)
```
Also, comment describing the conditional could shift down so it's a little closer to the code it describes. I also prefer when comments more closely match what is happening in the logic. So, instead of:
```
only run a test if it is the isolated test or if it is a test that the isolated test depends on passing
```
it could read:
```
skip the specified test if it is not the named isolated test and the isolated test is not dependent on the test to run.
```